File: day1/part1/solution.rb

def solve(rotations)
  current_position = 50
  zero_count = 0

  rotations.each do |line|
    line = line.strip
    next if line.empty?

    direction = line[0]
    distance = line[1..-1].to_i

    if direction == 'R'
      current_position += distance
    elsif direction == 'L'
      current_position -= distance
    end
    
    current_position %= 100

    if current_position == 0
      zero_count += 1
    end
  end
  zero_count
end

def main
  input_file_path = File.join(__dir__, 'input1.txt')
  rotations = File.readlines(input_file_path).map(&:strip) # Read lines and strip whitespace

  password = solve(rotations)
  puts "The password is: #{password}"
end

# Call the main function when the script is executed
main if __FILE__ == $PROGRAM_NAME

File: day1/part1/solution.exs
defmodule Day1Part1 do
  def solve(input_path) do
    File.stream!(input_path)
    |> Enum.reduce({50, 0}, fn line, {current_position, zero_count} ->
      line = String.trim(line)
      if String.length(line) > 0 do
        {direction, distance_str} = String.split_at(line, 1)
        distance = String.to_integer(distance_str)

        new_position =
          case direction do
            "R" -> current_position + distance
            "L" -> current_position - distance
          end

        new_position = rem(new_position, 100)

        new_position = if new_position < 0, do: new_position + 100, else: new_position

        new_zero_count =
          if new_position == 0 do
            zero_count + 1
          else
            zero_count
          end

        {new_position, new_zero_count}
      else
        {current_position, zero_count}
      end
    end)
    |> elem(1)
  end
end

password = Day1Part1.solve("day1/part1/input1.txt")
IO.puts("The password is: #{password}")